src/main/java/org/spdx/sbom/gradle/SpdxSbomPlugin.java

```diff
@@ -128,55 +110,33 @@ private void createTaskForTarget(
 
-                    Configuration pomsConfig =
-                        project
-                            .getConfigurations()
-                            .detachedConfiguration(
-                                project
-                                    .getConfigurations()
-                                    .getByName(configurationName)
-                                    .getIncoming()
-                                    .getResolutionResult()
-                                    .getAllComponents()
-                                    .stream()
-                                    .filter(rcr -> rcr.getId() instanceof ModuleComponentIdentifier)
-                                    .map(rcr -> rcr.getId().getDisplayName() + "@pom")
-                                    .map(pom -> project.getDependencies().create(pom))
-                                    .toArray(Dependency[]::new));
-                    t.getPoms()
-                        .putAll(PomResolver.newPomResolver(project).effectivePoms(pomsConfig));
-
                     t.getRootComponents().add(rootComponent);
                   }
                   t.getMavenRepositories()
                       .set(
                           project.provider(
                               () ->
-                                  project.getRepositories().getAsMap().entrySet().stream()
+                                  getAllRepositories(project).entrySet().stream()
                                       .filter(e -> e.getValue() instanceof MavenArtifactRepository)
-                                      .map(Entry::getKey)
                                       .collect(
                                           Collectors.toMap(
-                                              e -> e,
-                                              e ->
-                                                  ((MavenArtifactRepository)
-                                                          project.getRepositories().getByName(e))
-                                                      .getUrl()))));
+                                              Entry::getKey,
+                                              e -> ((MavenArtifactRepository) e.getValue()).getUrl()))));
                 });
     aggregate.configure(t -> t.dependsOn(task));
   }
 
-  private static class ArtifactTransformer
-      implements Transformer<
-          Map<ComponentArtifactIdentifier, File>, Collection<ResolvedArtifactResult>> {
+  private Map<String, ArtifactRepository> getAllRepositories(Project project) {
+    Map<String, ArtifactRepository> projectRepositories = project.getRepositories().getAsMap();
+
+    // nasty case because of https://github.com/gradle/gradle/issues/17295
+    Map<String, ArtifactRepository> settingsRepositories = ((GradleInternal) project.getGradle())
+        .getSettings()
+        .getDependencyResolutionManagement()
+        .getRepositories()
+        .getAsMap();
```

Comment:
I only use `dependencyResolutionManagement.repositories` to add my repositories, so the project one is empty. There could technically be repositories added in both places, so that's why I'm doing this.